examples/decoupled/repeat_model.py

```diff
@@ -0,0 +1,206 @@
+        Implementing `initialize` function is optional. This function allows
+        the model to intialize any state associated with this model.
+
+        Parameters
+        ----------
+        args : dict
+          Both keys and values are strings. The dictionary keys and values are:
+          * model_config: A JSON string containing the model configuration
+          * model_instance_kind: A string containing model instance kind
+          * model_instance_device_id: A string containing model instance device ID
+          * model_repository: Model repository path
+          * model_version: Model version
+          * model_name: Model name
+        """
+
+        # You must parse model_config. JSON string is not parsed here
+        self.model_config = model_config = json.loads(args['model_config'])
+
+        using_decoupled = pb_utils.using_decoupled_model_transaction_policy(
+            model_config)
+        if not using_decoupled:
+            raise pb_utils.TritonModelException(
+                """the model `{}` can generate any number of responses per request,
+                enable decoupled transaction policy in model configuration to 
+                serve this model""".format(args['model_name']))
+
+        # Get OUT configuration
+        out_config = pb_utils.get_output_config_by_name(model_config, "OUT")
+
+        # Get IDX configuration
+        idx_config = pb_utils.get_output_config_by_name(model_config, "IDX")
+
+        # Convert Triton types to numpy types
+        self.out_dtype = pb_utils.triton_string_to_numpy(
+            out_config['data_type'])
+        self.idx_dtype = pb_utils.triton_string_to_numpy(
+            idx_config['data_type'])
+
+        # To keep track of response threads so that we can delay
+        # the finalizing the model until all response threads
+        # have completed.
+        self.inflight_thread_count = 0
+        self.inflight_thread_count_lck = threading.Lock()
+
+    def execute(self, requests):
+        """`execute` MUST be implemented in every Python model. `execute`
+        function receives a list of pb_utils.InferenceRequest as the only
+        argument. This function is called when an inference request is made
+        for this model. The request.get_response_sender() must be used to
+        get an InferenceResponseSender object associated with the request.
+        Use the InferenceResponseSender.send() and InferenceResponseSender.close()
+        calls to send responses and indicate no responses will be sent for
+        the corresponding request respectively. If there is an error, you can
+        set the error argument when creating a pb_utils.InferenceResponse
+
+        Parameters
+        ----------
+        requests : list
+          A list of pb_utils.InferenceRequest
+
+        Returns
+        -------
+        None
+        """
+
+        # This model does not support batching, so 'request_count' should always be 1.
+        if len(requests) != 1:
```

Comment:
Is this check required? With `max_batch_size: 0` the requests size will be always 1.

Reply:
request size will not be 1 only when using dynamic batching. Anyways, this check makes it clear the model is not a batching model. And as the config is a separate file than this model.py, I think it makes it more comprehensible.

Reply:
> request size will not be 1 only when using dynamic batching. 

Can you clarify a little bit? Are you saying that the requests size can be larger than 1 even with max_batch_size equal to zero?

Reply:
No.. requests size will be one if max_batch_size == 0. 
I am saying it will also be 1 for max_batch_size > 1. 
The request size will not be 1 if max_batch_size > 1 and dynamic batching is enabled.
I have added this test to make it explicit that the model is not a batching model. So, it will fail if it is tried to be used with config.pbtxt setting max_batch_size > 1 and enabling dynamic batching. A user can change the config.pbtxt as per their needs and this check only makes sure that model will return an appropriate error.